html-aam/names.html

```diff
@@ -0,0 +1,52 @@
+  <script src="/resources/testharnessreport.js"></script>
+  <script src="/resources/testdriver.js"></script>
+  <script src="/resources/testdriver-vendor.js"></script>
+  <script src="/resources/testdriver-actions.js"></script>
+  <script src="/wai-aria/scripts/aria-utils.js"></script>
+</head>
+<body>
+
+
+<p>Tests the accName for elements defined in <a href="https://w3c.github.io/html-aam/#accessible-name-and-description-computation">HTML-AAM: Accessible Name Computations By HTML Element</a>. These tests are meant to show whether an element returns a name per the naming mechanism used. See <a href="https://wpt.fyi/results/accname">wpt: accname</a> for expanded accName testing.</p>
+
+
+<!-- element to reference for aria-labelledby tests -->
+<div id=labelledby hidden>labelledby</div>
+
+<address data-testname="address no name" data-expectedname="" class="ex">x</address>
+
+<address data-testname="address aria-label" aria-label="label" data-expectedname="label" class="ex">x</address>
+<address data-testname="address aria-labelledby" aria-labelledby="labelledby" data-expectedname="labelledby" class="ex">x</address>
+<address data-testname="address title" title="title" data-expectedname="title" class="ex">x</address>
+
+<address data-testname="address aria-labelledby vs aria-label vs title" aria-labelledby="labelledby" aria-label="label" title="title" data-expectedname="labelledby" class="ex">x</address>
+<address data-testname="address aria-labelledby vs aria-label" aria-labelledby="labelledby" aria-label="label" data-expectedname="labelledby" class="ex">x</address>
+
+<address data-testname="address aria-labelledby vs title" aria-labelledby="labelledby" title="title" data-expectedname="labelledby" class="ex">x</address>
+<address data-testname="address aria-label vs title" aria-label="label" title="title" data-expectedname="label" class="ex">x</address>
+
+
+
+<search data-testname="search no name" data-expectedname="" class="ex">x</search>
+
+<search data-testname="search aria-label" aria-label="label" data-expectedname="label" class="ex">x</search>
+<search data-testname="search aria-labelledby" aria-labelledby="labelledby" data-expectedname="labelledby" class="ex">x</search>
+<search data-testname="search title" title="title" data-expectedname="title" class="ex">x</search>
+
+<search data-testname="search aria-labelledby vs aria-label vs title" aria-labelledby="labelledby" aria-label="label" title="title" data-expectedname="labelledby" class="ex">x</search>
+<search data-testname="search aria-labelledby vs aria-label" aria-labelledby="labelledby" aria-label="label" data-expectedname="labelledby" class="ex">x</search>
+
+<search data-testname="search aria-labelledby vs title" aria-labelledby="labelledby" title="title" data-expectedname="labelledby" class="ex">x</search>
+<search data-testname="search aria-label vs title" aria-label="label" title="title" data-expectedname="label" class="ex">x</search>
+
```

Comment:
Should this file also include tests comparing `<img alt>` and `for/id` to `labelledby` and `label`? What about `table/caption`, `details/summary`, others? If there is overlap with [`/accname/name/comp_host_language_label.html`](https://github.com/web-platform-tests/wpt/blob/master/accname/name/comp_host_language_label.html) then we can move the HTML tests here, and add comment pointers between the files.

Reply:
i had orignally thought that elements that can only be named by author would be on one file, and then subsequent files could be created for the elements which have many more naming methods.

mostly though, i just wanted to make sure this was actually setup correctly / this was in the right folder for these (and not in accname?  i think it makes sense here though per the scope of what html aam is defining).

so if this is going in the right direction, then i can add on from here.

Reply:
but more specifically to your question / reviewing that file (and others) - yes i would have made those tests / put them over here (well in a different file) since they're specifically HTML element and not explicit ARIA roles.  There are also other varients i'd end up adding to those that i don't see in there (button named via a label element) or are in other files (via a title attr).

(i actually have a number of these shoved away in my own testing files that i can polish off and get into this format if this is generally going in the direction you think it should)

Reply:
Yeah direction is spot on. I don’t care if it lives in AccName or HTML-AAM, as long as there’s a comment in one cross-referencing the other.

Reply:
@cookiecrook is the reference to https://wpt.fyi/results/accname what you wanted for a cross ref, or a different page/url to mention?

Reply:
Yes... What you've added is fine. I like in-line cross-references like [this line in one of the accname tests](https://github.com/web-platform-tests/wpt/blob/master/accname/name/comp_host_language_label.html#L143) 

```html
<!-- SVG: -> /svg-aam/name/ -->
```
which points out the SVG tests that could be added here are actually already tested elsewhere. My hope is that it will keep us (and other contributors) from duplicating the same tests in multiple locations...

Reply:
Or [another example](https://github.com/web-platform-tests/wpt/blob/master/html-aam/roles.html#L30):

```html
<!-- bdi -> ./roles-generic.html -->
```

Effectively, "If you're reading this, don't add another `bdi` test here in `/html-aam/roles.html` b/c it's already tested in `/html-aam/roles-generic.html`"